pkg/controller/template/render.go

```diff
@@ -239,7 +254,15 @@ func generateMachineConfigForName(config *RenderConfig, role, name, path string)
 		return nil, fmt.Errorf("error transpiling ct config to Ignition config: %v", err)
 	}
 
-	return MachineConfigFromIgnConfig(role, name, ignCfg), nil
+	mcfg := MachineConfigFromIgnConfig(role, name, ignCfg)
+	osUpdatesEnabled, err := osUpdatesEnabledForRole(config, role)
+	if err != nil {
+		return nil, err
+	} else if osUpdatesEnabled {
```

Comment:
No need for this else, you're returning on the branch aboce anyway

Reply:
I'm confused, the other one is usual error handling?  Are you saying it'd e.g. be clearer like this?

```
diff --git a/pkg/controller/template/render.go b/pkg/controller/template/render.go
index 40aa9a8..97cf0e4 100644
--- a/pkg/controller/template/render.go
+++ b/pkg/controller/template/render.go
@@ -258,7 +258,8 @@ func generateMachineConfigForName(config *RenderConfig, role, name, path string)
 	osUpdatesEnabled, err := osUpdatesEnabledForRole(config, role)
 	if err != nil {
 		return nil, err
-	} else if osUpdatesEnabled {
+	}
+	if osUpdatesEnabled {
 		mcfg.Spec.OSImageURL = config.OSImageURL
 	}
 
```

Reply:
Yup, that's how it's usually done when you have an if branch which returns